src/main/java/com/iridium/iridiumskyblock/IridiumSkyblock.java

```diff
@@ -181,7 +181,11 @@ public void run() {
                 if (!islands.hasNext()) {
                     islands = getDatabaseManager().getIslandTableManager().getEntries().stream().map(Island::getId).collect(Collectors.toList()).listIterator();
                 } else {
-                    getIslandManager().getIslandById(islands.next()).ifPresent(island -> getIslandManager().recalculateIsland(island));
+                    getIslandManager().getIslandById(islands.next()).ifPresent(island -> {
+                        getIslandManager().recalculateIsland(island);
+                        island.getPlayersOnIsland().clear();
+                        Bukkit.getOnlinePlayers().stream().filter(player -> island.isInIsland(player.getLocation())).forEach(island.getPlayersOnIsland()::add);
```

Comment:
I mean... Just imagine a server with 5000 islands and 50 online players... The number of times you call this every 10 ticks would be insane

Reply:
in default configs this task runs every 10 seconds i think its fine but if you have any idea please lemme

Reply:
I actually just got an idea, just loop through all online players and add them to the player list of the island they are on

Reply:
but we must remove they from other islands

Reply:
Peaches added a cache recently, maybe you'll be able to use it. Otherwise, there is a great Caching system in the Google Guava library that allows entries to expire after a specific amount of time

Reply:
i've tried with cache thing but isn't fine (we can't get list from cache for adding or removing with moveevent), can you send a example/link for Guava lib ?

Reply:
https://github.com/google/guava/wiki/CachesExplained